src/odemis/util/img.py

```diff
@@ -466,8 +465,28 @@ def rescale_hq(data, shape):
       information that is linked to the size (e.g, pixel size), it is also
       updated.
     """
-    # TODO: support RGB(A) images
     # TODO: make it faster
+
+    # get the shape with all the dimensions
+    if len(data.shape) > 2:
```

Comment:
Just make sure the shape is the same as the array.shape (and drop all the cleverness about dimensions, which should be done beforehand).